pkg/network/usm/ebpf_gotls.go

```diff
@@ -307,9 +308,28 @@ func registerCBCreator(mgr *manager.Manager, offsetsDataMap *ebpf.Map, probeIDs
 	}
 }
 
+// DetachPID detaches the provided PID from the eBPF program.
+func (p *goTLSProgram) DetachPID(pid uint32) error {
+	return p.registry.Unregister(pid)
+}
+
+func (p *goTLSProgram) handleProcessExit(pid pid) {
+	_ = p.DetachPID(pid)
+}
+
 func (p *goTLSProgram) handleProcessStart(pid pid) {
+	_ = p.AttachPID(pid)
+}
+
+var (
+	errSelfExcluded                = errors.New("self-excluded")
+	errInternalDDogProcessRejected = errors.New("internal datadog process rejected")
+)
+
+// AttachPID attaches the provided PID to the eBPF program.
+func (p *goTLSProgram) AttachPID(pid uint32) error {
```

Comment:
nit: move all public methods to the top and leave the private ones at the bottom